application/src/File/Thumbnailer/Vips.php

```diff
@@ -0,0 +1,229 @@
+            $this->setVipsPath($this->getOption('vips_dir'));
+        }
+    }
+
+    public function create($strategy, $constraint, array $options = [])
+    {
+        if ($this->vipsPath === false) {
+            throw new Exception\CannotCreateThumbnailException;
+        }
+
+        if ($this->getIsOldVips()) {
+            return $this->createWithOldVips($strategy, $constraint, $options);
+        }
+
+        $origPath = $this->source;
+
+        // Available parameters on load.
+        $mediaType = $this->sourceFile->getMediaType();
+        $supportPages = [
+            'application/pdf',
+            'image/gif',
+            'image/tiff',
+            'image/webp',
+        ];
+        $supportDpi = [
+            'application/pdf',
+            'image/svg+xml',
+        ];
+        $supportBackground = [
+            'application/pdf',
+        ];
+        // Other via magick: page, density.
+        $loadOptions = [];
+        if (in_array($mediaType, $supportPages)) {
+            $page = (int) $this->getOption('page', 0);
+            $loadOptions[] = "page=$page";
+        }
+        if (in_array($mediaType, $supportDpi)) {
+            $loadOptions[] = 'dpi=150';
+        }
+        if (in_array($mediaType, $supportBackground)) {
+            $loadOptions[] = 'background=255 255 255';
+        }
+        if (count($loadOptions)) {
+            $origPath .= '[' . implode(',', $loadOptions) . ']';
+        }
+
+        if ($strategy === 'square') {
+            $vipsCrop = [
+                // "none" does not crop (default).
+                // 'none',
+                'low',
+                'centre',
+                'high',
+                'attention',
+                'entropy',
+                // "all" does not crop as square.
+                // 'all',
+            ];
+            $mapImagickToVips = [
+                'northwest' => 'high',
+                'north' => 'high',
+                'northearth' => 'high',
+                'west' => 'centre',
+                'center' => 'centre',
+                'east' => 'centre',
+                'southwest' => 'low',
+                'south' => 'low',
+                'southeast' => 'low',
+            ];
+            $gravity = isset($options['gravity']) ? strtolower($options['gravity']) : 'attention';
+            if (isset($mapImagickToVips[$gravity])) {
+                $gravity = $mapImagickToVips[$gravity];
+            } elseif (!in_array($gravity, $vipsCrop)) {
+                $gravity = 'attention';
+            }
+            $crop = ' --crop ' . $gravity;
+        } else {
+            $crop = '';
+        }
+
+        $tempFile = $this->tempFileFactory->build();
+        $tempPath = $tempFile->getTempPath() . '.jpg';
+        $tempPathCommand = $tempPath . '[background=255 255 255,optimize-coding]';
+        $tempFile->delete();
+
+        $command = sprintf(
+            '%s thumbnail %s %s %d --height %d%s%s --size both --linear --intent absolute',
```

Comment:
From my understanding, "both" isn't appropriate for "size" here: the other thumbnailers don't scale up, only down (except for the "square" type).

Reply:
This is fixed in last commit.